cpp/dolfinx/fem/DirichletBC.cpp

```diff
@@ -107,6 +169,8 @@ get_remote_bcs1(const common::IndexMap& map,
   const std::vector<std::int64_t>& ghosts = map.ghosts();
 
   // Build map from ghost to local position
+  // NOTE: Should we use map here or just one vector with ghosts and
```

Comment:
Sorting takes `O(n * log(n))` , which is the same cost as the creation of the map.
`map.find` and `lower_bound`  are both `O(log(n)` . 

So if the ghosts were sorted I would use a vector intead of a map.

Reply:
Currently ghosts are not sorted, so I guess this is a thing we can improve if we change the indexmap, ref #1816